### PR TITLE
Super pipeline KFPv2.

### PR DESCRIPTION
**Pull request by Mohammad-nassar10**

### CODE REVIEW

File: kfp/superworkflows/ray/kfp_v2/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+- It is not required to upload the transforms pipelines before running the superpipeline.
+- It created just one run that includes all the nested transfroms and their sub-tasks.
+- No need for additional component as `executeSubWorkflowComponent.yaml`. All the implementation in the same pipeline file.
+- One missing thing in this implementation is the overriding parameters option.
+
+### How to compile the superpipeline
+```
+export KFPv2=1
+export PYTHONPATH=../../../../transforms
+python3 superpipeline_noop_docId_v2.py
```
Comment (revit13) [inline kfp/superworkflows/ray/kfp_v2/README.md]: I think setting the venv with the libraries is missing

File: kfp/kfp_ray_components/createRayClusterComponent.yaml [hunk 2]
```diff
@@ -4,14 +4,14 @@ description: Creates Ray cluster
...
-    - { name: ray_worker_options, type: String, default: "", description: "ray worker configuration" }
+    - { name: ray_head_options, type: JsonObject, default: "{}", description: "ray head configuration" }
+    - { name: ray_worker_options, type: JsonObject, default: "{}", description: "ray worker configuration" }
     - { name: server_url, type: String, default: "", description: "url of api server" }
     - { name: additional_params, type: String, default: "{}", description: "additional parameters" }
 
 implementation:
     container:
-        image: "quay.io/dataprep1/data-prep-kit/kfp-data-processing:latest"
+        image: "quay.io/dataprep1/data-prep-kit/kfp-data-processing_v2:latest"
```
Comment (revit13) [inline kfp/kfp_ray_components/createRayClusterComponent.yaml]: No need to change the default image name

File: kfp/superworkflows/ray/kfp_v2/README.md (hunk 1 above)
Comment (revit13) [reply kfp/superworkflows/ray/kfp_v2/README.md]: also a step to compile the workflow which will set the versions (update kfp image to v2)

File: transforms/universal/doc_id/kfp_ray/doc_id_wf.py [hunk 3]
```diff
@@ -22,16 +22,16 @@
 # the name of the job script
 EXEC_SCRIPT_NAME: str = "doc_id_transform_ray.py"
 # components
-base_kfp_image = "quay.io/dataprep1/data-prep-kit/kfp-data-processing:latest"
+base_kfp_image = "quay.io/dataprep1/data-prep-kit/kfp-data-processing_v2:latest"
```
Comment (revit13) [inline transforms/universal/doc_id/kfp_ray/doc_id_wf.py]: No need to change the default image name

File: transforms/universal/noop/kfp_ray/noop_wf.py [hunk 4]
```diff
@@ -23,22 +23,22 @@
 EXEC_SCRIPT_NAME: str = "noop_transform_ray.py"
 
 # components
-base_kfp_image = "quay.io/dataprep1/data-prep-kit/kfp-data-processing:latest"
+base_kfp_image = "quay.io/dataprep1/data-prep-kit/kfp-data-processing_v2:latest"
```
Comment (revit13) [inline transforms/universal/noop/kfp_ray/noop_wf.py]: No need to change the default image name

File: kfp/kfp_ray_components/createRayClusterComponent.yaml (hunk 2 above)
Comment (Mohammad-nassar10) [reply kfp/kfp_ray_components/createRayClusterComponent.yaml]: Fixed.

File: transforms/universal/doc_id/kfp_ray/doc_id_wf.py (hunk 3 above)
Comment (Mohammad-nassar10) [reply transforms/universal/doc_id/kfp_ray/doc_id_wf.py]: Fixed.

File: transforms/universal/noop/kfp_ray/noop_wf.py (hunk 4 above)
Comment (Mohammad-nassar10) [reply transforms/universal/noop/kfp_ray/noop_wf.py]: Fixed.

File: kfp/superworkflows/ray/kfp_v2/README.md (hunk 1 above)
Comment (Mohammad-nassar10) [reply kfp/superworkflows/ray/kfp_v2/README.md]: Done.

**Review by revit13 — APPROVED** [review]

File: kfp/superworkflows/ray/kfp_v2/README.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+![noop_layer_pipeline](noop_layer.png)
+
+Another useful feature of the KFP v2 is the `Json` editor for the `dict` type input parameter as shown here:
+![json_param](json_param.png)
+
+## Main differences from KFP v1 superpipeline:
+- It is not required to upload the transforms pipelines before running the superpipeline.
+- It created just one run that includes all the nested transfroms and their sub-tasks.
+- No need for additional component as `executeSubWorkflowComponent.yaml`. All the implementation in the same pipeline file.
+- One missing thing in this implementation is the overriding parameters option.
```
Comment (roytman) [inline kfp/superworkflows/ray/kfp_v2/README.md]: From here it is not clear what do you mean

File: kfp/superworkflows/ray/kfp_v2/README.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+![noop_nested_pipeline](noop_nested.png)
+
+`root -> noop-ray-pipeline -> exit-handler-1` Layer
+![noop_layer_pipeline](noop_layer.png)
+
+Another useful feature of the KFP v2 is the `Json` editor for the `dict` type input parameter as shown here:
+![json_param](json_param.png)
+
+## Main differences from KFP v1 superpipeline:
+- It is not required to upload the transforms pipelines before running the superpipeline.
```
Comment (roytman) [inline kfp/superworkflows/ray/kfp_v2/README.md]: maybe to add that the entire pipeline and the nested pipelines are up to date.

File: kfp/superworkflows/ray/kfp_v2/README.md (hunk 5 above)
Comment (Mohammad-nassar10) [reply kfp/superworkflows/ray/kfp_v2/README.md]: Is this sentence better:
`In superpipelines of KFP v1 there exists an option to override the common parameters with specific values for each one of the transforms. This option is missing in the KFP v2 superpipelines.`

**Review by roytman — APPROVED** [review]

**Pull request merged**